### PR TITLE
DDF-1648 - added itests for ApplicationService and ordered install pr…

### PR DESCRIPTION
**Pull request by kcwire**
…ofiles
@stustison 
@pklinef 
@coyotesqrl 
@bdeining

### CODE REVIEW

File: distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/ServiceManager.java [hunk 1]
```diff
@@ -369,6 +370,62 @@ public void waitForRequiredBundles(String symbolicNamePrefix) throws Interrupted
         }
     }
 
+    public void waitForRequiredFeature(String featureName) throws Exception {
```
Comment (pklinef) [inline distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/ServiceManager.java]: You can extract the common code from these two methods and have them pass in a `Predicate<FeatureState>` to the common method for the test logic.

https://docs.oracle.com/javase/8/docs/api/java/util/function/Predicate.html

File: distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/platform/TestApplicationService.java [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+                status.getState(),
+                is(ACTIVE));
+
+        Application solr = applicationService.getApplication(SOLR_APP);
+        assertNotNull("Application [" + SOLR_APP + "] must not be null", solr);
+        status = applicationService.getApplicationStatus(solr);
+        assertThat("Application [" + SOLR_APP + "] should be INACTIVE",
+                status.getState(),
+                is(INACTIVE));
+
```
Comment (coyotesqrl) [inline distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/platform/TestApplicationService.java]: For better code coverage, should probably add one more test that invokes `applicationService.getApplications()` and pulls out the catalog and solr apps (or two others) and checks their status that way.

File: distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/platform/TestApplicationService.java [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+        URI sdkUri = sdkApp.getURI();
+
+        // Remove
+        applicationService.removeApplication(sdkApp);
+        Set<Application> apps = applicationService.getApplications();
+        assertThat(apps, not(hasItem(sdkApp)));
+
+        // Add
+        applicationService.addApplication(sdkUri);
+        sdkApp = applicationService.getApplication(SDK_APP);
```
Comment (coyotesqrl) [inline distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/platform/TestApplicationService.java]: Might be better to call `getApplications` here again.

File: distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/platform/TestApplicationService.java (hunk 2 above)
Comment (kcwire) [inline distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/platform/TestApplicationService.java]: Updated this method to get the apps from the the `getApplications()` method instead.

File: distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/ServiceManager.java (hunk 1 above)
Comment (kcwire) [inline distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/ServiceManager.java]: yeah I don't know what I was thinking - fixed as suggested.

File: platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/impl/ApplicationServiceImpl.java [hunk 4]
```diff
@@ -720,6 +748,30 @@ public void stopApplication(String application) throws ApplicationServiceExcepti
...
+            BundleStateSet bundleStates = getCurrentBundleStates(application.getFeatures());
+            if (!bundleStates.getTransitionalBundles()
+                    .isEmpty()) {
+                if (System.currentTimeMillis() > timeoutLimit) {
+                    break;
+                }
+                LOGGER.trace("Waiting for the following bundles to become ACTIVE: {}",
+                        bundleStates.getTransitionalBundles()
+                                .stream()
+                                .map(i -> i.toString())
```
Comment (coyotesqrl) [inline platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/impl/ApplicationServiceImpl.java]: Can convert to a method reference.

**Pull request merged**